From b2f808bea9910d552e5ba34f6b12244d98b669fd Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 21 Sep 2023 05:04:48 +0300
Subject: [PATCH] [FSW-402] fs1: add register map and mask

This patch ports regmap and regmask for fs1. These
offsets and masks are based on new specs directory
structure.

Change-Id: Iba7447cae079890bf7794c338385d5fcaf144305
---

diff --git a/arch/fs1/arc/regmap.h b/arch/fs1/arc/regmap.h
new file mode 100644
index 0000000..b85d0b5
--- /dev/null
+++ b/arch/fs1/arc/regmap.h
@@ -0,0 +1,142 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2023 HabanaLabs Ltd.
+ */
+
+#ifndef _FS1_REGMAP_H_
+#define _FS1_REGMAP_H_
+
+/*
+ * NOTE: All register offsets used for regmap_readl & regmap_writel should be of
+ * IO DIE0 as the api will perform address manipulation for other dies.
+ * Hardware performs address manipulation only for PARC_LOCAL_CFG page in configuration
+ * space where DIE ID register is available.
+ */
+#define FW_BLOCK_BASE(SPECS_BLOCK_BASE)     (SPECS_BLOCK_BASE & 0xF0000000)
+
+#define mmPARC_LOCAL_DIE_ID		(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_ARC_MNG_LOCAL_BASE) \
+					+ mmPSOC_ARC_WRAP_LOCAL_DIE_CPU_ID)
+#define mmI2C_S_BASE			(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P0_I2C_S_BASE))
+#define mmI2C_S_IC_CON			(mmI2C_S_BASE + mmP0_I2C_S_IC_CON)
+#define mmI2C_S_IC_SAR			(mmI2C_S_BASE + mmP0_I2C_S_IC_SAR)
+#define mmI2C_S_IC_INTR_MASK		(mmI2C_S_BASE + mmP0_I2C_S_IC_INTR_MASK)
+#define mmI2C_S_IC_RX_TL		(mmI2C_S_BASE + mmP0_I2C_S_IC_RX_TL)
+#define mmI2C_S_IC_TX_TL		(mmI2C_S_BASE + mmP0_I2C_S_IC_TX_TL)
+#define mmI2C_S_IC_ENABLE		(mmI2C_S_BASE + mmP0_I2C_S_IC_ENABLE)
+
+#define mmQSPI_BASE			(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_QSPI_BASE))
+#define mmQSPI_SER			(mmQSPI_BASE + mmP1_QSPI_SER)
+#define mmQSPI_BAUDR			(mmQSPI_BASE + mmP1_QSPI_BAUDR)
+#define mmQSPI_DR0			(mmQSPI_BASE + mmP1_QSPI_DR0)
+
+#define mmEFUSE_BASE            (FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P0_EFUSE_BASE))
+#define mmEFUSE_READ_CTRL       (mmEFUSE_BASE + mmP0_EFUSE_RD)
+#define mmEFUSE_READ_ADDRESS    (mmEFUSE_BASE + mmP0_EFUSE_READ_ADDRESS)
+#define mmEFUSE_DATA_0			(mmEFUSE_BASE + mmP0_EFUSE_DATA_0)
+
+#define mmTIMESTAMP_BASE        (FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P0_TIMESTAMP_BASE))
+#define mmTIMESTAMP_CNTCR	(mmTIMESTAMP_BASE + mmP0_TIMESTAMP_CNTCR)
+#define mmTIMESTAMP_CNTSR	(mmTIMESTAMP_BASE + mmP0_TIMESTAMP_CNTSR)
+#define mmTIMESTAMP_CNTCVL	(mmTIMESTAMP_BASE + mmP0_TIMESTAMP_CNTCVL)
+#define mmTIMESTAMP_CNTCVU	(mmTIMESTAMP_BASE + mmP0_TIMESTAMP_CNTCVU)
+#define mmTIMESTAMP_CNTFID0	(mmTIMESTAMP_BASE + mmP0_TIMESTAMP_CNTFID0)
+
+#define mmPRSTN                 (FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+				+ mmP1_MISC_PRSTN)
+#define mmCOLD_RST_FLOPS_2	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_BOOT_BASE)\
+				+ mmIO_PSOC_CFG_BOOT_COLD_RST_REG_2)
+#define mmPCIE_PHY_RST_N	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+				+ mmP1_MISC_PCIE_PHY_RST_N)
+#define mmBOOT_STRAP_PINS_L	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_BOOT_BASE) \
+				+ mmIO_PSOC_CFG_BOOT_BOOT_STRAP_PINS_L)
+#define mmPRSTN_MASK		(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_BOOT_BASE) \
+				 + mmIO_PSOC_CFG_BOOT_PRSTN_RST_MASK)
+
+#define mmSPI_DMA_AWUSER	\
+	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_PA_MISC_SPY_DMA_HBW_ATTR_CTRL_BASE) \
+	+ mmPARC_GLOBAL_CONF_SPY_DMA_HBW_ATTR_CTRL_HB_MMU_BYPASS_OVRD)
+#define mmSPI_DMA_CTRL		(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_PA_MISC_BASE) \
+				+ mmPA_MISC_SPI_DMA_CTRL)
+#define mmSPI_DMA_STATUS	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_PA_MISC_BASE) \
+				+ mmPA_MISC_SPI_DMA_STATUS)
+#define mmSPI_DMA_DST_ADDR_L	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_PA_MISC_BASE) \
+				+ mmPA_MISC_SPI_DMA_DST_ADDR_L)
+#define mmSPI_DMA_DST_ADDR_H	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_PA_MISC_BASE) \
+				+ mmPA_MISC_SPI_DMA_DST_ADDR_H)
+
+#define mmMEM_CPY_CTRL		(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+				+ mmP1_MISC_MEM_CPY_CTRL)
+#define mmMEM_CPY_STATUS	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+				+ mmP1_MISC_MEM_CPY_STATUS)
+#define mmMEM_CPY_DEST_ADDR_H	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+				+ mmP1_MISC_MEM_CPY_DEST_ADDR_H)
+#define mmMEM_CPY_DEST_ADDR_L	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+				+ mmP1_MISC_MEM_CPY_DEST_ADDR_L)
+#define mmMEM_CPY_CTRL2		(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+				+ mmP1_MISC_MEM_CPY_CTRL2)
+#define mmMEM_CPY_AXUSER \
+	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_INTR_CTRL_MEM_CPY_HBW_ATTR_CTRL_BASE) \
+	+ mmPSOC_GLOBAL_CONF2_MEM_CPY_HBW_ATTR_CTRL_HB_MMU_BYPASS_OVRD)
+
+#define mmSPI_DIRECT_WR_RD_CTRL		(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+					+ mmP1_MISC_SPI_DIRECT_WR_RD_CTRL)
+#define mmSYSCFG1_EFUSE_DATA		(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+					+ mmP1_MISC_SCRATCHPAD_0)
+#define mmSYSCFG2_EFUSE_DATA		(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+					+ mmP1_MISC_SCRATCHPAD_1)
+#define mmFFUSE_WRITE_IND		(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_MISC_BASE) \
+					+ mmP1_MISC_SCRATCHPAD_2)
+
+#define mmGPIO0_DATA			(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_GPIO0_BASE) \
+					+ mmIO_PSOC_CFG_GPIO0_DATA)
+
+#define mmSECURITY_BASE			(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_SECURITY_BASE))
+#define mmSECURITY_MIN_FW_VERSION	(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_MIN_FW_VERSION)
+#define mmSECURITY_PRI_IMG_STATUS	(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_PRI_IMG_STATUS)
+#define mmSECURITY_SEC_IMG_STATUS	(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SEC_IMG_STATUS)
+#define mmSECURITY_PRI_IMG_FW_VER	(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_PRI_IMG_FW_VER)
+#define mmSECURITY_SEC_IMG_FW_VER	(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SEC_IMG_FW_VER)
+#define mmSECURITY_STATUS	        (mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_STATUS)
+#define mmSECURITY_KEY_REVOKE		(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_KEY_REVOKE)
+#define mmSW_STATUS_0			(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SW_STATUS_0)
+#define mmSW_STATUS_1			(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SW_STATUS_1)
+#define mmSW_STATUS_2			(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SW_STATUS_2)
+#define mmSW_STATUS_3			(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SW_STATUS_3)
+#define mmSW_STATUS_4			(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SW_STATUS_4)
+#define mmSW_STATUS_5			(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SW_STATUS_5)
+#define mmSW_STATUS_6			(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SW_STATUS_6)
+#define mmSW_STATUS_7			(mmSECURITY_BASE + mmIO_PSOC_CFG_SECURITY_SW_STATUS_7)
+
+
+/* Using V1 offset file. */
+#define mmJT_MSTR_IF_CORE_HBW_AXPROT_OVD	\
+	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P0_JT_MSTR_IF_AXPROT_HBW_BASE) \
+	 + mmMSTR_IF_V1_AXPROT_HBW_ARPROT_0_OVRD)
+
+#define mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVD	\
+	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P1_SMI_MSTR_IF_AXPROT_HBW_BASE) \
+	+ mmMSTR_IF_V1_AXPROT_HBW_ARPROT_0_OVRD)
+
+#define mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVD	\
+	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_0_PSOC_P0_I2C_S_MSTR_IF_AXPROT_HBW_BASE) \
+	+ mmMSTR_IF_V1_AXPROT_HBW_ARPROT_0_OVRD)
+
+/* TODO: All below registers are taken from CXL EP instead of PCIe EP. */
+#define mmCPU_HOT_RST	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_1_CXL_EP_PCIE_CORE_BASE) \
+			+ mmCXL_EP_CFG_PCIE_CORE_HOT_RST)
+#define mmPCIE_RSP_OK	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_1_CXL_EP_PCIE_WRAP_BASE) \
+			+ mmCXL_EP_CFG_PCIE_WRAP_PCIE_RSP_OK_0)
+
+#define mmLINK_CAPABILITIES_REG		\
+	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_1_CXL_EP_DBI_SIG_BASE) \
+	+ mmCXL_EP_CFG_DBI_SIG_LINK_CAPABILITIES_REG)
+#define mmLINK_CONTROL2_LINK_STATUS2_REG	\
+	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_1_CXL_EP_DBI_SIG_BASE) \
+	+ mmCXL_EP_CFG_DBI_SIG_LINK_CTRL2_LINK_STATUS2_REG)
+
+#define mmLTSSM_EN	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_1_CXL_EP_PCIE_AUX_BASE) \
+			+ mmCXL_EP_CFG_PCIE_AUX_LTSSM_EN)
+#define mmAPP_SRIS_MODE	(FW_BLOCK_BASE(mmIO_0_CFG_HALF_1_CXL_EP_PCIE_AUX_BASE) \
+			+ mmCXL_EP_CFG_PCIE_AUX_APP_SRIS_MODE)
+
+#endif /* _FS1_REGMAP_H_ */
diff --git a/arch/fs1/arc/regmask.h b/arch/fs1/arc/regmask.h
new file mode 100644
index 0000000..fb4f27f
--- /dev/null
+++ b/arch/fs1/arc/regmask.h
@@ -0,0 +1,53 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2023 HabanaLabs Ltd.
+ */
+
+#ifndef _FS1_REGMASK_H_
+#define _FS1_REGMASK_H_
+
+#define TIMESTAMP_CNTCR_CNTCR_HDBG_M		P0_TIMESTAMP_CNTCR_CNTCR_HDBG_M
+#define TIMESTAMP_CNTCR_CNTCR_EN_M		P0_TIMESTAMP_CNTCR_CNTCR_EN_M
+
+#define SPI_DMA_CTRL_ADDR_SHIFT			PA_MISC_SPI_DMA_CTRL_ADDR_SHIFT_S
+#define SPI_DMA_CTRL_DST_SRAM_SHIFT		PA_MISC_SPI_DMA_CTRL_DST_SRAM_S
+#define SPI_DMA_STATUS_ERROR_MASK		PA_MISC_SPI_DMA_STATUS_ERROR_M
+#define SPI_DMA_STATUS_ERROR_SHIFT		PA_MISC_SPI_DMA_STATUS_ERROR_S
+#define SPI_DMA_STATUS_DONE_MASK		PA_MISC_SPI_DMA_STATUS_DONE_M
+#define SPI_DMA_CTRL_FLD_SIZE_SHIFT		PA_MISC_SPI_DMA_CTRL_MEM_SIZE_S
+#define SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK	\
+					P1_MISC_SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_M
+#define PRSTN_MASK_IND_MASK			IO_PSOC_CFG_BOOT_PRSTN_RST_MASK_IND_M
+#define MEM_CPY_CTRL2_USE_CONST_SHIFT		P1_MISC_MEM_CPY_CTRL2_USE_CONST_S
+
+#define SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_SHIFT	\
+					IO_PSOC_CFG_SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_S
+#define SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_SHIFT	\
+					IO_PSOC_CFG_SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_S
+#define SECURITY_PRI_IMG_STATUS_CRC_FAIL_SHIFT	IO_PSOC_CFG_SECURITY_PRI_IMG_STATUS_CRC_FAIL_S
+#define SECURITY_STATUS_IMG_VALIDATED_SHIFT	COMP_PSOC_CFG_SECURITY_STATUS_IMG_VALIDATED_S
+#define SECURITY_STATUS_FW_UPGRADE_SHIFT	IO_PSOC_CFG_SECURITY_STATUS_FW_UPGRADE_S
+#define SECURITY_STATUS_KEY_REVOKED_SHIFT	IO_PSOC_CFG_SECURITY_STATUS_KEY_REVOKED_S
+#define SECURITY_STATUS_KEY_REVOKED_IND_SHIFT	IO_PSOC_CFG_SECURITY_STATUS_KEY_REVOKED_IND_S
+#define SECURITY_STATUS_MIN_FW_MODIFIED_SHIFT	IO_PSOC_CFG_SECURITY_STATUS_MIN_FW_MODIFIED_S
+#define SECURITY_STATUS_FW_VER_INC_SHIFT	IO_PSOC_CFG_SECURITY_STATUS_FW_VER_INC_S
+#define SECURITY_STATUS_SPI_IMG_MODIFIES_SHIFT	IO_PSOC_CFG_SECURITY_STATUS_SPI_IMG_MODIFIES_S
+#define SECURITY_STATUS_FW_AUTHENTICAION_FAIL_SHIFT	\
+					IO_PSOC_CFG_SECURITY_STATUS_FW_AUTHENTICAION_FAIL_S
+
+#define LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_MASK	\
+				CXL_EP_CFG_DBI_SIG_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_M
+#define LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_SHIFT	\
+				CXL_EP_CFG_DBI_SIG_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_S
+#define LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_MASK	\
+			CXL_EP_CFG_DBI_SIG_LINK_CTRL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_M
+#define LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_SHIFT	\
+			CXL_EP_CFG_DBI_SIG_LINK_CTRL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_S
+
+#define LTSSM_EN_IND_MASK			CXL_EP_CFG_PCIE_AUX_LTSSM_EN_IND_M
+
+#define CPU_HOT_RST_CTRL_MASK			CXL_EP_CFG_PCIE_CORE_HOT_RST_INTR_M
+
+#define FORCE_RECOVERY_MASK			BIT(1)	//GPIO_01
+
+#endif /* _FS1_REGMASK_H_ */
